public/app/assetlibrary/item/item.html

```diff
@@ -24,23 +24,53 @@
   <div class="col-pane">
     <!-- PREVIEW -->
     <div data-ng-if="asset.type !== 'whiteboard'">
-      <div class="text-center assetlibrary-item-preview" id="assetlibrary-item-preview" data-ng-show="previewStatus === 'complete'"></div>
-      <div class="text-center assetlibrary-item-preview-message" data-ng-if="previewStatus === 'pending'">
+
+      <!-- COMPLETED PREVIEWS -->
+      <div class="text-center assetlibrary-item-preview" id="assetlibrary-item-preview" data-ng-show="asset.preview_status === 'done'">
+        <!-- Documents -->
+        <div data-ng-if="asset.type === 'file' && asset.pdf_url !== null">
+          <iframe class="preview-document" width="100%" height="800" data-ng-src="{{asset.embedUrl}}" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen="" allowscriptaccess="always" scrolling="no"></iframe>
+        </div>
+
+        <!-- Images -->
+        <div data-ng-if="asset.type === 'file' && asset.image_url !== null">
+          <img class="preview-image" alt="{{asset.title}}" title="{{asset.title}}" src="{{asset.image_url}}" style="max-width: 100%; max-height: inherit; height: inherit; width: inherit;" />
+        </div>
+
+        <!-- Links -->
+        <div data-ng-if="asset.type === 'link' && asset.isEmbeddable">
+          <iframe width="100%" height="800" src="{{asset.embedUrl}}" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen="" allowscriptaccess="always" scrolling="no"></iframe>
+        </div>
+
+        <div data-ng-if="asset.type === 'link' && asset.preview_metadata.youtubeId !== undefined">
+          <iframe width="100%" height="800" src="{{asset.embedUrl}}" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen="" allowscriptaccess="always" scrolling="no"></iframe>
+        </div>
+
+        <div data-ng-if="asset.type === 'link' && !asset.isEmbeddable && asset.image_url">
```

Comment:
Do we need a fallback for link that are not embeddable and we weren't able to generate an image for ?